File: csharp/242.ValidAnagram.cs
namespace Leetcode.ValidAnagram;

public class Solution {
    public bool IsAnagram(string s, string t) =>
      s.Length == t.Length &&
      s
        .GroupBy(letter => letter)
        .All(group => s.Count(letter => letter == group.Key) == t.Count(letter => letter == group.Key));
}

File: golang/242.ValidAnagram.go
package leetcode

func isAnagram(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}

	letters := make(map[byte]int, len(s))
	for i := range s {
		letters[s[i]]++
		letters[t[i]]--
	}
	for _, v := range letters {
		if v != 0 {
			return false
		}
	}
	return true
}
